README.md

```diff
@@ -107,6 +107,15 @@ The successful response contains data in `response.data`. The following example
 response.data.create_invoice.first.id
 ```
 
+### Schema caching and dumping
+
+To reduce requests to graphql API you can cache schema:
+
+```ruby
+client = Client.new(url, schema_path: 'config/your_graphql_schema.json')
+client.dump_schema # you only need to call this when graphql schema changes
```

Comment:
I think this should be a bang method since it has side effects, so `dump_schema!`.

Reply:
it depends which style is preferred in graphlient. In my own projects I prefer not to use prima dona methods (https://github.com/troessner/reek/blob/master/docs/Prima-Donna-Method.md). 

I like idea with `schema.dump`, but we will be forced to change class of `schema` instance which can cause backwards incompatibility. Do you think it is worth the risk?

Reply:
You should try and see how it feels. I don't feel too strongly about it.

Reply:
If you call `client.schema.dump!` for the first time it'd involve network and disk IO, which is a good indicator for adding a `!`. In order to keep the compatibility, you could use a delegator object:

```ruby
require 'delegate'

def schema
  @schema ||= SchemaDelegate.new(GraphQL::Client.load_schema(http))
end

class SchemaDelegator < SimpleDelegator
  def dump!
    ...
  end
end

private_constant :SchemaDelegator
```

Reply:
I used `SimpleDelegator` as you suggested, but I moved Schema to separate file. So it means that we can't use `private_constant`. If that's an issue I can move it back in to `client.rb`